tests/test_marc.py

```diff
@@ -109,6 +110,34 @@ def test_add_web_client_urls(self):
         # Web client URLs can come from either the MARC export integration or
         # a library registry integration.
 
+        # The expected format for a web client url is as follows
+        # - <web-client-base>/book/<cm-base>/<lib-short-name>/works/<qualified-identifier>
+        client_url_template = "{client_base}/book/{cm_base}/{lib}/works/{qid}"
```

Comment:
@kristojorg, @leonardr This is the format for the "long" urls. Can you confirm that this is what is needed?

Reply:
It looks right to me but Kristo's in a position to verify this.

Reply:
Everything after `/book/` should be encoded as a url parameter. And I'm not sure I've seen the `/lib` section before. This is an example of a url:
```
http://localhost:3000/lyrasis/book/https%3A%2F%2Fdemo.lyrasistechnology.org%2Flyrasis%2Fworks%2FURI%2Fhttps%3A%2F%2Fwww.feedbooks.com%2Fitem%2F2936784
```
If you decode everything after `/book/` you end up with:
```
"https://demo.lyrasistechnology.org/lyrasis/works/URI/https://www.feedbooks.com/item/2936784"
```

Reply:
Oh, duh, the `{lib}` is the library short name. I think this works

Reply:
Thanks @kristojorg. I'm currently encoding only the qualified identifier (identifier.type + "/" + identifier.identifier), since that is the only part that could have special characters. Here's a kind of fake example for what is produced from one of the test cases:

- http://web_catalog/book/http://test-circulation-manager/default/works/Gutenberg%20ID%2Fidentifier

Does that work for you? Or do you need me to encode the entire CM endpoint again (effectively double encoding the identifier)?

Reply:
This actually doesn't work in the current CPW. The reason is that this url:
```
http://localhost:3000/lyrasis/book/https://demo.lyrasistechnology.org/lyrasis/works/URI%2Fhttps%3A%2F%2Fwww.feedbooks.com%2Fitem%2F2936784
```
Is telling the application to look for a page at `/lyrasis/book/https://demo.lyrasistechnology.org/lyrasis/works/`, which doesn't exist. The page that does exist is `/lyrasys/book/:bookUrl`. 

Basically, it does need to be double encoded because it is actually a url embedded in a url. This is a valid url: `http://test-circulation-manager/default/works/Gutenberg%20ID%2Fidentifier`, but if you want to embed that in the CPW url, you need to encode it again since in that context, the forward slashes are in fact special characters. Does that make any sense?